File: jira/frontend/src/types/notification.ts
// ============================================================================
// Notification Types
// ============================================================================

// ── Enums / Union types ─────────────────────────────────────────────────────

export type NotificationType =
  | "task_assigned"
  | "comment_added"
  | "status_changed"
  | "mention"
  | "sprint_started"
  | "sprint_completed";

// ── Entity ──────────────────────────────────────────────────────────────────

export interface Notification {
  id: string;
  user_id: string;
  type: NotificationType;
  title: string;
  message: string;
  entity_type: string | null;
  entity_id: string | null;
  is_read: boolean;
  created_at: string;
}

// ── Request payloads ────────────────────────────────────────────────────────

export interface MarkNotificationReadRequest {
  notification_ids: string[];
}

// ── Query helpers ───────────────────────────────────────────────────────────

export interface NotificationQueryParams {
  page?: number;
  page_size?: number;
  is_read?: boolean;
  type?: NotificationType;
}

File: project-hub/backend/app/schemas/notification.py
from pydantic import BaseModel


class NotificationResponse(BaseModel):
    id: str
    user_id: str
    type: str
    title: str
    message: str
    is_read: bool
    entity_type: str | None = None
    entity_id: str | None = None
    created_at: str

    model_config = {"from_attributes": True}


class UnreadCountResponse(BaseModel):
    count: int
